tests/test_serving.py

```diff
@@ -278,3 +278,21 @@ def app(environ, start_response):
         'content_type': 'application/json'
     })
     assert r.content == b'YES'
+
+
+def test_port_must_be_integer(dev_server):
+    with pytest.raises(TypeError) as port_type_exception_with_reloader:
+        def app(environ, start_response):
```

Comment:
You could define this `app` method in the outer scope (i.e  not in the context manager) to reuse in your 2 tests.